File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        ans = l = maxf = 0 
        count = {}

        for r in range(len(s)):
            count[s[r]] = count.get(s[r],0) + 1
            maxf = max(maxf, count[s[r]])
        
            while r - l + 1 - maxf > k:
                count[s[l]] -= 1
                l += 1
            ans = max(r - l + 1, ans)
        
        return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        if(nums == null || nums.Length == 0)
            return 0;
        if(nums.Length == 1) return 1 ;
        var list = new HashSet<int>(nums).ToList<int>() ;
        list.Sort() ;
        int ans = 1 ;
        int max = 1 ;
        int? pre = null ;
        for(int i =0 ; i< list.Count; i++){
            if(pre == null){
                pre = list[i] ;
            }else{
                if(pre + 1 == list[i]){
                    ans ++ ;
                    max = Math.Max(max, ans) ;
                }else{
                    ans = 1 ;
                }
                pre = list[i] ;
            }
        }
        return max ;
    }
}
